### PR TITLE
configurable failurePolicy for validating webhooks

### PR DESCRIPTION
**Pull request by attiss**
This PR adds a new configuration option to the helm chart: `crds.validationFailurePolicy`. The value is injected into the validating webhook's `failurePolicy` field. Defaults to `Fail` (current behavior).

---

Closes https://github.com/metallb/metallb/issues/1598

### CODE REVIEW

**Comment by attiss** [comment]
# test: default value

<details>
<summary><code>helm install --dry-run --debug metallb .</code></summary>

```
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/attiss/git/metallb/charts/metallb

NAME: metallb
LAST DEPLOYED: Mon Oct  3 19:34:10 2022
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
controller:
  affinity: {}
  enabled: true
  image:
    pullPolicy: null
    repository: quay.io/metallb/controller
    tag: null
  livenessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  logLevel: info
  nodeSelector: {}
  podAnnotations: {}
  priorityClassName: ""
  readinessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  resources: {}
  runtimeClassName: ""
  securityContext:
    fsGroup: 65534
    runAsNonRoot: true
    runAsUser: 65534
  serviceAccount:
    annotations: {}
    create: true
    name: ""
  strategy:
    type: RollingUpdate
  tolerations: []
crds:
  enabled: true
  global: {}
  validationFailurePolicy: Fail
fullnameOverride: ""
imagePullSecrets: []
loadBalancerClass: ""
nameOverride: ""
prometheus:
  controllerMetricsTLSSecret: ""
  metricsPort: 7472
  namespace: ""
  podMonitor:
    additionalLabels: {}
    annotations: {}
    enabled: false
    interval: null
    jobLabel: app.kubernetes.io/name
    metricRelabelings: []
    relabelings: []
  prometheusRule:
    additionalLabels: {}
    addressPoolExhausted:
      enabled: true
      labels:
        severity: alert
    addressPoolUsage:
      enabled: true
      thresholds:
      - labels:
          severity: warning
        percent: 75
      - labels:
          severity: warning
        percent: 85
      - labels:
          severity: alert
        percent: 95
    annotations: {}
    bgpSessionDown:
      enabled: true
      labels:
        severity: alert
    configNotLoaded:
      enabled: true
      labels:
        severity: warning
    enabled: false
    extraAlerts: []
    staleConfig:
      enabled: true
      labels:
        severity: warning
  rbacPrometheus: true
  rbacProxy:
    repository: gcr.io/kubebuilder/kube-rbac-proxy
    tag: v0.12.0
  scrapeAnnotations: false
  serviceAccount: ""
  serviceMonitor:
    controller:
      additionalLabels: {}
      annotations: {}
      tlsConfig:
        insecureSkipVerify: true
    enabled: false
    interval: null
    jobLabel: app.kubernetes.io/name
    metricRelabelings: []
    relabelings: []
    speaker:
      additionalLabels: {}
      annotations: {}
      tlsConfig:
        insecureSkipVerify: true
  speakerMetricsTLSSecret: ""
rbac:
  create: true
speaker:
  affinity: {}
  enabled: true
  frr:
    enabled: false
    image:
      pullPolicy: null
      repository: frrouting/frr
      tag: v7.5.1
    metricsPort: 7473
  image:
    pullPolicy: null
    repository: quay.io/metallb/speaker
    tag: null
  livenessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  logLevel: info
  memberlist:
    enabled: true
    mlBindPort: 7946
  nodeSelector: {}
  podAnnotations: {}
  priorityClassName: ""
  readinessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  resources: {}
  runtimeClassName: ""
  serviceAccount:
    annotations: {}
    create: true
    name: ""
  tolerateMaster: true
  tolerations: []
  updateStrategy:
    type: RollingUpdate

HOOKS:
MANIFEST:
<output omitted>
---
# Source: metallb/templates/webhooks.yaml
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  name: metallb-webhook-configuration
  labels:
    helm.sh/chart: metallb-0.0.0
    app.kubernetes.io/name: metallb
    app.kubernetes.io/instance: metallb
    app.kubernetes.io/version: "v0.0.0"
    app.kubernetes.io/managed-by: Helm
webhooks:
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-addresspool
  failurePolicy: Fail
  name: addresspoolvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - addresspools
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta2-bgppeer
  failurePolicy: Fail
  name: bgppeervalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta2
    operations:
    - CREATE
    - UPDATE
    resources:
    - bgppeers
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-ipaddresspool
  failurePolicy: Fail
  name: ipaddresspoolvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - ipaddresspools
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-bgpadvertisement
  failurePolicy: Fail
  name: bgpadvertisementvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - bgpadvertisements
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-community
  failurePolicy: Fail
  name: communityvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - communities
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-bfdprofile
  failurePolicy: Fail
  name: bfdprofilevalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - DELETE
    resources:
    - bfdprofiles
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-l2advertisement
  failurePolicy: Fail
  name: l2advertisementvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - l2advertisements
  sideEffects: None

NOTES:
MetalLB is now running in the cluster.

Now you can configure it via its CRs. Please refer to the metallb official docs
on how to use the CRs.
```

</details>

**Comment by attiss** [comment]
# test: `Fail`

<details>
<summary><code>helm install --dry-run --debug --set crds.validationFailurePolicy=Fail metallb .</code></summary>

```
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/attiss/git/metallb/charts/metallb

NAME: metallb
LAST DEPLOYED: Mon Oct  3 19:29:42 2022
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
crds:
  validationFailurePolicy: Fail

COMPUTED VALUES:
controller:
  affinity: {}
  enabled: true
  image:
    pullPolicy: null
    repository: quay.io/metallb/controller
    tag: null
  livenessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  logLevel: info
  nodeSelector: {}
  podAnnotations: {}
  priorityClassName: ""
  readinessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  resources: {}
  runtimeClassName: ""
  securityContext:
    fsGroup: 65534
    runAsNonRoot: true
    runAsUser: 65534
  serviceAccount:
    annotations: {}
    create: true
    name: ""
  strategy:
    type: RollingUpdate
  tolerations: []
crds:
  enabled: true
  global: {}
  validationFailurePolicy: Fail
fullnameOverride: ""
imagePullSecrets: []
loadBalancerClass: ""
nameOverride: ""
prometheus:
  controllerMetricsTLSSecret: ""
  metricsPort: 7472
  namespace: ""
  podMonitor:
    additionalLabels: {}
    annotations: {}
    enabled: false
    interval: null
    jobLabel: app.kubernetes.io/name
    metricRelabelings: []
    relabelings: []
  prometheusRule:
    additionalLabels: {}
    addressPoolExhausted:
      enabled: true
      labels:
        severity: alert
    addressPoolUsage:
      enabled: true
      thresholds:
      - labels:
          severity: warning
        percent: 75
      - labels:
          severity: warning
        percent: 85
      - labels:
          severity: alert
        percent: 95
    annotations: {}
    bgpSessionDown:
      enabled: true
      labels:
        severity: alert
    configNotLoaded:
      enabled: true
      labels:
        severity: warning
    enabled: false
    extraAlerts: []
    staleConfig:
      enabled: true
      labels:
        severity: warning
  rbacPrometheus: true
  rbacProxy:
    repository: gcr.io/kubebuilder/kube-rbac-proxy
    tag: v0.12.0
  scrapeAnnotations: false
  serviceAccount: ""
  serviceMonitor:
    controller:
      additionalLabels: {}
      annotations: {}
      tlsConfig:
        insecureSkipVerify: true
    enabled: false
    interval: null
    jobLabel: app.kubernetes.io/name
    metricRelabelings: []
    relabelings: []
    speaker:
      additionalLabels: {}
      annotations: {}
      tlsConfig:
        insecureSkipVerify: true
  speakerMetricsTLSSecret: ""
rbac:
  create: true
speaker:
  affinity: {}
  enabled: true
  frr:
    enabled: false
    image:
      pullPolicy: null
      repository: frrouting/frr
      tag: v7.5.1
    metricsPort: 7473
  image:
    pullPolicy: null
    repository: quay.io/metallb/speaker
    tag: null
  livenessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  logLevel: info
  memberlist:
    enabled: true
    mlBindPort: 7946
  nodeSelector: {}
  podAnnotations: {}
  priorityClassName: ""
  readinessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  resources: {}
  runtimeClassName: ""
  serviceAccount:
    annotations: {}
    create: true
    name: ""
  tolerateMaster: true
  tolerations: []
  updateStrategy:
    type: RollingUpdate

HOOKS:
MANIFEST:
<output omitted>
---
# Source: metallb/templates/webhooks.yaml
apiVersion: v1
kind: Service
metadata:
  name: metallb-webhook-service
  labels:
    helm.sh/chart: metallb-0.0.0
    app.kubernetes.io/name: metallb
    app.kubernetes.io/instance: metallb
    app.kubernetes.io/version: "v0.0.0"
    app.kubernetes.io/managed-by: Helm
spec:
  ports:
  - port: 443
    targetPort: 9443
  selector:
    app.kubernetes.io/name: metallb
    app.kubernetes.io/instance: metallb
    app.kubernetes.io/component: controller
---
# Source: metallb/templates/speaker.yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: metallb-speaker
  labels:
    helm.sh/chart: metallb-0.0.0
    app.kubernetes.io/name: metallb
    app.kubernetes.io/instance: metallb
    app.kubernetes.io/version: "v0.0.0"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: speaker
spec:
  updateStrategy:
    type: RollingUpdate
  selector:
    matchLabels:
      app.kubernetes.io/name: metallb
      app.kubernetes.io/instance: metallb
      app.kubernetes.io/component: speaker
  template:
    metadata:
      labels:
        app.kubernetes.io/name: metallb
        app.kubernetes.io/instance: metallb
        app.kubernetes.io/component: speaker
    spec:
      serviceAccountName: metallb-speaker
      terminationGracePeriodSeconds: 0
      hostNetwork: true
      volumes:
      containers:
      - name: speaker
        image: quay.io/metallb/speaker:v0.0.0
        args:
        - --port=7472
        - --log-level=info
        env:
        - name: METALLB_NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
        - name: METALLB_HOST
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
        - name: METALLB_ML_BIND_ADDR
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        - name: METALLB_ML_LABELS
          value: "app.kubernetes.io/name=metallb,app.kubernetes.io/component=speaker"
        - name: METALLB_ML_BIND_PORT
          value: "7946"
        - name: METALLB_ML_SECRET_KEY
          valueFrom:
            secretKeyRef:
              name: metallb-memberlist
              key: secretkey
        ports:
        - name: monitoring
          containerPort: 7472
        - name: memberlist-tcp
          containerPort: 7946
          protocol: TCP
        - name: memberlist-udp
          containerPort: 7946
          protocol: UDP
        livenessProbe:
          httpGet:
            path: /metrics
            port: monitoring
          initialDelaySeconds: 10
          periodSeconds: 10
          timeoutSeconds: 1
          successThreshold: 1
          failureThreshold: 3
        readinessProbe:
          httpGet:
            path: /metrics
            port: monitoring
          initialDelaySeconds: 10
          periodSeconds: 10
          timeoutSeconds: 1
          successThreshold: 1
          failureThreshold: 3
        securityContext:
          allowPrivilegeEscalation: false
          readOnlyRootFilesystem: true
          capabilities:
            drop:
            - ALL
            add:
            - NET_RAW
      nodeSelector:
        "kubernetes.io/os": linux
      tolerations:
      - key: node-role.kubernetes.io/master
        effect: NoSchedule
        operator: Exists
      - key: node-role.kubernetes.io/control-plane
        effect: NoSchedule
        operator: Exists
---
# Source: metallb/templates/controller.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: metallb-controller
  labels:
    helm.sh/chart: metallb-0.0.0
    app.kubernetes.io/name: metallb
    app.kubernetes.io/instance: metallb
    app.kubernetes.io/version: "v0.0.0"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: controller
spec:
  strategy:
    type: RollingUpdate
  selector:
    matchLabels:
      app.kubernetes.io/name: metallb
      app.kubernetes.io/instance: metallb
      app.kubernetes.io/component: controller
  template:
    metadata:
      labels:
        app.kubernetes.io/name: metallb
        app.kubernetes.io/instance: metallb
        app.kubernetes.io/component: controller
    spec:
      serviceAccountName: metallb-controller
      terminationGracePeriodSeconds: 0
      securityContext:
        fsGroup: 65534
        runAsNonRoot: true
        runAsUser: 65534
      containers:
      - name: controller
        image: quay.io/metallb/controller:v0.0.0
        args:
        - --port=7472
        - --log-level=info
        - --cert-service-name=metallb-webhook-service
        env:
        - name: METALLB_ML_SECRET_NAME
          value: metallb-memberlist
        - name: METALLB_DEPLOYMENT
          value: metallb-controller
        ports:
        - name: monitoring
          containerPort: 7472
        - containerPort: 9443
          name: webhook-server
          protocol: TCP
        volumeMounts:
        - mountPath: /tmp/k8s-webhook-server/serving-certs
          name: cert
          readOnly: true
        livenessProbe:
          httpGet:
            path: /metrics
            port: monitoring
          initialDelaySeconds: 10
          periodSeconds: 10
          timeoutSeconds: 1
          successThreshold: 1
          failureThreshold: 3
        readinessProbe:
          httpGet:
            path: /validate-metallb-io-v1beta1-addresspool
            port: webhook-server
            scheme: HTTPS
          initialDelaySeconds: 10
          periodSeconds: 10
          timeoutSeconds: 1
          successThreshold: 1
          failureThreshold: 3
        securityContext:
          allowPrivilegeEscalation: false
          readOnlyRootFilesystem: true
          capabilities:
            drop:
            - ALL
      nodeSelector:
        "kubernetes.io/os": linux
      volumes:
      - name: cert
        secret:
          defaultMode: 420
          secretName: webhook-server-cert
---
# Source: metallb/templates/webhooks.yaml
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  name: metallb-webhook-configuration
  labels:
    helm.sh/chart: metallb-0.0.0
    app.kubernetes.io/name: metallb
    app.kubernetes.io/instance: metallb
    app.kubernetes.io/version: "v0.0.0"
    app.kubernetes.io/managed-by: Helm
webhooks:
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-addresspool
  failurePolicy: Fail
  name: addresspoolvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - addresspools
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta2-bgppeer
  failurePolicy: Fail
  name: bgppeervalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta2
    operations:
    - CREATE
    - UPDATE
    resources:
    - bgppeers
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-ipaddresspool
  failurePolicy: Fail
  name: ipaddresspoolvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - ipaddresspools
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-bgpadvertisement
  failurePolicy: Fail
  name: bgpadvertisementvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - bgpadvertisements
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-community
  failurePolicy: Fail
  name: communityvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - communities
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-bfdprofile
  failurePolicy: Fail
  name: bfdprofilevalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - DELETE
    resources:
    - bfdprofiles
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-l2advertisement
  failurePolicy: Fail
  name: l2advertisementvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - l2advertisements
  sideEffects: None

NOTES:
MetalLB is now running in the cluster.

Now you can configure it via its CRs. Please refer to the metallb official docs
on how to use the CRs.
```

</details>

**Comment by attiss** [comment]
# test: `Ignore`

<details>
<summary><code>helm install --dry-run --debug --set crds.validationFailurePolicy=Ignore metallb .</code></summary>

```
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/attiss/git/metallb/charts/metallb

NAME: metallb
LAST DEPLOYED: Mon Oct  3 19:30:45 2022
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
crds:
  validationFailurePolicy: Ignore

COMPUTED VALUES:
controller:
  affinity: {}
  enabled: true
  image:
    pullPolicy: null
    repository: quay.io/metallb/controller
    tag: null
  livenessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  logLevel: info
  nodeSelector: {}
  podAnnotations: {}
  priorityClassName: ""
  readinessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  resources: {}
  runtimeClassName: ""
  securityContext:
    fsGroup: 65534
    runAsNonRoot: true
    runAsUser: 65534
  serviceAccount:
    annotations: {}
    create: true
    name: ""
  strategy:
    type: RollingUpdate
  tolerations: []
crds:
  enabled: true
  global: {}
  validationFailurePolicy: Ignore
fullnameOverride: ""
imagePullSecrets: []
loadBalancerClass: ""
nameOverride: ""
prometheus:
  controllerMetricsTLSSecret: ""
  metricsPort: 7472
  namespace: ""
  podMonitor:
    additionalLabels: {}
    annotations: {}
    enabled: false
    interval: null
    jobLabel: app.kubernetes.io/name
    metricRelabelings: []
    relabelings: []
  prometheusRule:
    additionalLabels: {}
    addressPoolExhausted:
      enabled: true
      labels:
        severity: alert
    addressPoolUsage:
      enabled: true
      thresholds:
      - labels:
          severity: warning
        percent: 75
      - labels:
          severity: warning
        percent: 85
      - labels:
          severity: alert
        percent: 95
    annotations: {}
    bgpSessionDown:
      enabled: true
      labels:
        severity: alert
    configNotLoaded:
      enabled: true
      labels:
        severity: warning
    enabled: false
    extraAlerts: []
    staleConfig:
      enabled: true
      labels:
        severity: warning
  rbacPrometheus: true
  rbacProxy:
    repository: gcr.io/kubebuilder/kube-rbac-proxy
    tag: v0.12.0
  scrapeAnnotations: false
  serviceAccount: ""
  serviceMonitor:
    controller:
      additionalLabels: {}
      annotations: {}
      tlsConfig:
        insecureSkipVerify: true
    enabled: false
    interval: null
    jobLabel: app.kubernetes.io/name
    metricRelabelings: []
    relabelings: []
    speaker:
      additionalLabels: {}
      annotations: {}
      tlsConfig:
        insecureSkipVerify: true
  speakerMetricsTLSSecret: ""
rbac:
  create: true
speaker:
  affinity: {}
  enabled: true
  frr:
    enabled: false
    image:
      pullPolicy: null
      repository: frrouting/frr
      tag: v7.5.1
    metricsPort: 7473
  image:
    pullPolicy: null
    repository: quay.io/metallb/speaker
    tag: null
  livenessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  logLevel: info
  memberlist:
    enabled: true
    mlBindPort: 7946
  nodeSelector: {}
  podAnnotations: {}
  priorityClassName: ""
  readinessProbe:
    enabled: true
    failureThreshold: 3
    initialDelaySeconds: 10
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
  resources: {}
  runtimeClassName: ""
  serviceAccount:
    annotations: {}
    create: true
    name: ""
  tolerateMaster: true
  tolerations: []
  updateStrategy:
    type: RollingUpdate

HOOKS:
MANIFEST:
<output omitted>
---
# Source: metallb/templates/webhooks.yaml
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  name: metallb-webhook-configuration
  labels:
    helm.sh/chart: metallb-0.0.0
    app.kubernetes.io/name: metallb
    app.kubernetes.io/instance: metallb
    app.kubernetes.io/version: "v0.0.0"
    app.kubernetes.io/managed-by: Helm
webhooks:
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-addresspool
  failurePolicy: Ignore
  name: addresspoolvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - addresspools
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta2-bgppeer
  failurePolicy: Ignore
  name: bgppeervalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta2
    operations:
    - CREATE
    - UPDATE
    resources:
    - bgppeers
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-ipaddresspool
  failurePolicy: Ignore
  name: ipaddresspoolvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - ipaddresspools
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-bgpadvertisement
  failurePolicy: Ignore
  name: bgpadvertisementvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - bgpadvertisements
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-community
  failurePolicy: Ignore
  name: communityvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - communities
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-bfdprofile
  failurePolicy: Ignore
  name: bfdprofilevalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - DELETE
    resources:
    - bfdprofiles
  sideEffects: None
- admissionReviewVersions:
  - v1
  clientConfig:
    service:
      name: metallb-webhook-service
      namespace: default
      path: /validate-metallb-io-v1beta1-l2advertisement
  failurePolicy: Ignore
  name: l2advertisementvalidationwebhook.metallb.io
  rules:
  - apiGroups:
    - metallb.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - l2advertisements
  sideEffects: None

NOTES:
MetalLB is now running in the cluster.

Now you can configure it via its CRs. Please refer to the metallb official docs
on how to use the CRs.

```

</details>

**Comment by attiss** [comment]
# test: invalid value

<details>
<summary><code>helm install --dry-run --debug --set crds.validationFailurePolicy=Invalid metallb .</code></summary>

```
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/attiss/git/metallb/charts/metallb

Error: INSTALLATION FAILED: values don't meet the specifications of the schema(s) in the following chart(s):
metallb:
- crds.validationFailurePolicy: crds.validationFailurePolicy must be one of the following: "Ignore", "Fail"

helm.go:84: [debug] values don't meet the specifications of the schema(s) in the following chart(s):
metallb:
- crds.validationFailurePolicy: crds.validationFailurePolicy must be one of the following: "Ignore", "Fail"

INSTALLATION FAILED
main.newInstallCmd.func2
	helm.sh/helm/v3/cmd/helm/install.go:141
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.5.0/command.go:872
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.5.0/command.go:990
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.5.0/command.go:918
main.main
	helm.sh/helm/v3/cmd/helm/helm.go:83
runtime.main
	runtime/proc.go:250
runtime.goexit
	runtime/asm_amd64.s:1594
```

</details>

**Comment by fedepaol** [comment]
@attiss can you run inv helm-docs to regenerate the docs?
Thanks

File: charts/metallb/templates/webhooks.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ webhooks:
       name: metallb-webhook-service
       namespace: {{ .Release.Namespace }}
       path: /validate-metallb-io-v1beta1-addresspool
-  failurePolicy: Fail
+  failurePolicy: {{ default "Fail" .Values.crds.validationFailurePolicy }}
```
Comment (fedepaol) [inline charts/metallb/templates/webhooks.yaml:15]: I think there's no need to specify the default here, as long as we add it in values.yaml
same for the following.
cc @gclawes

**Comment by fedepaol** [comment]
Since you have to change the pr, can you adjust the commit subject and message following our reference https://chris.beams.io/posts/git-commit/ ?
Thanks!

File: charts/metallb/templates/webhooks.yaml (hunk 1 above)
Comment (attiss) [reply charts/metallb/templates/webhooks.yaml]: fair point, it's unnecessary duplication of the default value, updated

**Comment by attiss** [comment]
>Since you have to change the pr, can you adjust the commit subject and message following our reference https://chris.beams.io/posts/git-commit/ ?

Sure, updated! Thank you!

**Pull request merged**